crates/ra_ide/src/display/function_signature.rs

```diff
@@ -157,6 +162,22 @@ impl From<&'_ ast::FnDef> for FunctionSignature {
             res
         }
 
+        fn param_name_list(node: &ast::FnDef) -> Vec<String> {
+            let mut res = vec![];
+            if let Some(param_list) = node.param_list() {
+                if let Some(self_param) = param_list.self_param() {
+                    res.push(self_param.syntax().text().to_string())
+                }
+
+                res.extend(
+                    param_list
+                        .params()
+                        .map(|param| param.pat().unwrap().syntax().text().to_string()),
```

Comment:
This `unwarp` can fire: all nodes in the AST might be incomplete

Reply:
You’re right. Maybe I should make `parameter_name_list` a `Vec<Option<String>>` here?

Reply:
I'd do just `.unwrap_or_default()` for simplicity